frontend/src/views/Environment.tsx

```diff
@@ -9,8 +9,20 @@ import { ReactNode } from "react";
 import ReactMarkdown from 'react-markdown'
 import mermaid from 'mermaid'
 import TabControl from '../components/TabControl'
-const hostname = process.env.REACT_APP_API_HOST || ''
-const wsHostname = process.env.REACT_APP_WS_HOST || ''
+
+const protocol = window && window.location && window.location.protocol;
```

Comment:
You can use a combination of nullish coalescing operator and chaining operator to shorten that assigment a lot. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator

It might make sense to extract all HTTP requests and the setup for the endpoints into a dedicated file that can be consumed by multiple components if needed since the same setup was used in another component.

Reply:
@PBug90 Thanks for the hint! Did not know this nice construct! 

An abstraction for all HTTP requests might be too much. Implemented a new component that is now used by the four files needing to construct backend URLs. Seems more compact.